apps/newsletters-ui/src/app/components/NewsletterJsonEdit.tsx

```diff
@@ -0,0 +1,79 @@
+import { Alert, Snackbar, Typography } from '@mui/material';
+import { useState } from 'react';
+import type { NewsletterData } from '@newsletters-nx/newsletters-data-client';
+import { newsletterDataSchema } from '@newsletters-nx/newsletters-data-client';
+import { requestNewsletterEdit } from '../api-requests/request-newsletter-edit';
+import { JsonEditor } from './JsonEditor';
+
+interface Props {
+	originalItem: NewsletterData;
+}
+
+export const NewsletterJsonEdit = ({ originalItem }: Props) => {
+	const [item, setItem] = useState(originalItem);
+	const [showConfirmation, setShowConfirmation] = useState(false);
+	const [errorMessage, setErrorMessage] = useState<string | undefined>(
+		undefined,
+	);
+
+	const handleSubmission = async (record: NewsletterData) => {
+		const partial = {
+			...record,
+			listId: undefined,
+			identityName: undefined,
+		} as Partial<NewsletterData>;
+		const apiResonse = await requestNewsletterEdit(record.listId, partial);
+
+		if (apiResonse.ok) {
+			setItem(apiResonse.data);
+			setShowConfirmation(true);
```

Comment:
After a successful Submit, would probably be better to return to the previous screen (especially since Reset will no longer have any effect)